### DISCUSSION TITLE
Lazy load of router

### DISCUSSION THREAD

**Discussion by orfisko** [fastapi/fastapi#11724] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# No code
```

### Description

In order to shorten cold start times on AWS lambda, I was wondering whether it would be possible to write a middleware that would lazily load the router into the app depending on the path that is being transferred by the API gateway. I would love to hear whether it would be technically feasible and if so, if anyone has some ideas on how to realize this. 
Cold starts are a hot topic in the fastapi world, this could perhaps be a simple solution to it? 

### Operating System

Linux

### FastAPI Version

N/A

### Pydantic Version

N/A

### Python Version

N/A

**Comment by YuriiMotov** [comment]
@orfisko, have you managed to find a solution?

**Comment by orfisko** [comment]
No I haven't, I lack time to really dive into it. It's still a thing
though.

Op za 17 mei 2025, 08:16 schreef Motov Yurii ***@***.***>:
